maint/scripts/run_local_ci_test.sh

```
@@ -1,22 +1,197 @@
+# Normalize DEVICE to lowercase
+DEVICE=$(echo "$DEVICE" | tr 'A-Z' 'a-z')
+case "$DEVICE" in
+  cuda|hip|cpu|metal) ;;
+  *) echo "[ERROR] Unsupported DEVICE='$DEVICE'. Choose cuda|hip|cpu|metal." >&2; exit 2;;
+esac
+
 # Set ROOT_DIR to the project root (two levels up from this script's directory)
 ROOT_DIR=$(cd "$(dirname "$0")/../.." && pwd)
 
 # Change to the project root directory for local testing of changes
 cd $ROOT_DIR
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)